docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,339 @@
+| `ConsumerRewardsAllocationBytePrefix`   | Stores the ICS rewards per chain                                                                                             | consumer or provider chain       | provider chain                   | Every IBC transfer packet that sends rewards to the provider                                             |
+| `ConsumerCommissionRatePrefix`          | Commission rate per chain per validator                                                                                      | validator                        | consumer chain                   | Every `MsgSetConsumerCommissionRate` message                                                             |
+| `MinimumPowerInTopNBytePrefix`          | Stores the minimum power needed to opt in for a chain                                                                        | not applicable for Opt In chains | not applicable for Opt In chains | Every epoch                                                                                              |
+| `ConsumerAddrsToPruneV2BytePrefix`      | Stores consumer addresses to be pruned (as part of `VSCMaturedPacket`s deprecation)                                          | validator or provider chain      | provider chain                   | Every `MsgAssignConsumerKey` or `MsgOptIn` and later during actual pruning                               |
+
+Everything stored under one of the above keys is associated with a `chainId` and has to be migrated to new state under a `consumerId`.
+
+### New Messages
+In what follows, we describe the new messages (i.e., `MsgRegisterConsumerChain`, `MsgInitializeConsumerChain`, and `MsgUpdateConsumerChain`)
+that Permissionless ICS introduces, and on how those can be used.
+Then, we describe how to utilize these messages with our existing codebase.
+
+#### Register a Consumer Chain
+We first have to register an Opt In chain before launching it. This is done through the following message.
+```protobuf
+message MsgRegisterConsumerChain {
+  // the title of this launch 
+  string title;
+  // the description of this consumer chain
+  string description;
+  // the chain id of the new consumer chain
+  string chain_id;
+  // the owner of this consumer chain
+  string owner_address;
+}
+```
+
+This response of this message contains a single `string`, that is the `consumerId` for this registered consumer chain and initializes
+a consumer chain in its registered phase. With the returned `consumerId`, validators can already opt in on the consumer
+chain to show their potential interest on the chain. Additionally, a front-end ICS launchpad can also present
+this chain. Additionally, this allows consumer chains to show that they are interested in joining ICS even though,
```

Comment:
The front-end platform is orthogonal to this work and should not be part of this ADR. 
```suggestion
chain to show their potential interest on the chain. Additionally, this allows consumer chains to show that they are interested in joining ICS even though,
```